src/parsers/vct/parsers/transform/LLVMContractToCol.scala

```diff
@@ -0,0 +1,358 @@
+package vct.parsers.transform
+
+import org.antlr.v4.runtime.{ParserRuleContext, Token}
+import vct.antlr4.generated.LLVMSpecParser._
+import vct.antlr4.generated.LLVMSpecParserPatterns
+import vct.antlr4.generated.LLVMSpecParserPatterns._
+import vct.col.ast._
+import vct.col.origin.{ExpectedError, Origin, SourceNameOrigin}
+import vct.col.ref.{Ref, UnresolvedRef}
+import vct.col.util.AstBuildHelpers.{ff, foldAnd, implies, tt}
+
+import scala.annotation.nowarn
+import scala.collection.mutable
+
+@nowarn("msg=match may not be exhaustive&msg=Some\\(")
+case class LLVMContractToCol[G](override val originProvider: OriginProvider,
+                                override val blameProvider: BlameProvider,
+                                override val errors: Seq[(Token, Token, ExpectedError)],
+                                references: Seq[(String, Ref[G, Declaration[G]])])
+  extends ToCol(originProvider, blameProvider, errors) {
+  val referenceLut: mutable.Map[String, Ref[G, Declaration[G]]] = mutable.Map(references: _*)
```

Comment:
If i understand correctly, this would mean the following:
- introduce an `LLVMLocal` Node type
- Instead of directly resolving the reference in the `LLVMContractToCol` parser, introduce an `LLVMLocal`
- In the `LangLLVMToCol` rewriter, resolve the `LLVMLocal` to a variable.

A similair structure could apply to function invocations

Reply:
> It would be nice if you could take this out now that we decided we should do parsing from resolution: it is also consistent with the notion that there should be no logic or unexpected stuff in the antlr -> col transformatiosn.

Latest commit should resolve above issue. I'm planning to implement invocations in contracts soon too. Since it's kind of related I could add to this PR or open a separate PR for it.